plaso/analysis/windows_services.py

```diff
@@ -111,27 +111,28 @@ def __eq__(self, other_service):
     return True
 
   @classmethod
-  def FromEvent(cls, service_event):
-    """Creates a service object from an event.
+  def FromEventData(cls, event_data):
+    """Creates a service object from event data.
 
     Args:
-      service_event (EventObject): event to create a new service object from.
+      event_data (EventData): event data.
 
     Returns:
       WindowsService: service.
     """
-    _, _, name = service_event.key_path.rpartition(
+    _, _, name = event_data.key_path.rpartition(
         WindowsService._REGISTRY_KEY_PATH_SEPARATOR)
-    service_type = service_event.regvalue.get('Type', '')
-    image_path = service_event.regvalue.get('ImagePath', '')
-    start_type = service_event.regvalue.get('Start', '')
-    service_dll = service_event.regvalue.get('ServiceDll', '')
-    object_name = service_event.regvalue.get('ObjectName', '')
-
-    if service_event.pathspec:
-      source = (service_event.pathspec.location, service_event.key_path)
+    service_type = event_data.regvalue.get('Type', '')
+    image_path = event_data.regvalue.get('ImagePath', '')
+    start_type = event_data.regvalue.get('Start', '')
+    service_dll = event_data.regvalue.get('ServiceDll', '')
+    object_name = event_data.regvalue.get('ObjectName', '')
+
+    if event_data.pathspec:
```

Comment:
Pathspec is set on event, not event data

Reply:
see previous comment, it is set on event data but that might not be directly obvious from the code, and agree needs clean up